### ISSUE TITLE
StreamChatRN not working on monorepo setup

### ISSUE THREAD

**Issue by ericlifs** [GetStream/stream-chat-react-native#546]
**Describe the bug**
I have installed `stream-chat-react-native` version 3.0.0 and all it dependencies into my app which is architectured under a yarn workspaces monorepo and the chat screen is crashing as soon as it gets mounted due to a lot of Reanimated v2 errors. I have seen bunch of issues on their repo talking about this:

* https://github.com/software-mansion/react-native-reanimated/issues/858
* https://github.com/software-mansion/react-native-reanimated/issues/1720
* https://github.com/software-mansion/react-native-reanimated/issues/1508

As someone pointed out there, I have also verified by also installing Reanimated 2.0.0-rc.3 (which is still not documented) and still the same issues are happening for me. 

If I copy and paste my code to the Typescript example app and works perfectly.

**Expected behavior**
Stream Chat should work on yarn workspaces monorepos too.

**Additional context**
This is the exact error I'm getting:

```
2021-03-07 18:36:38.458151-0300 My App[42260:1132253] [native] Tried to synchronously call function {assign} from a different thread.
Occurred in worklet location: /Users/adsadsdas/workspace/adsads/packages/asdadsdas/node_modules/react-native-reanimated/src/reanimated2/animations.js (51:7)
Possible solutions are:
a) If you want to synchronously execute this method, mark it as a Worklet
b) If you want to execute this method on the JS thread, wrap it using runOnJS

libc++abi.dylib: terminating with uncaught exception of type facebook::jsi::JSError: undefined is not an object (evaluating 'delete animationCopy.callback')
decorateAnimation
[native code]
_f
[native code]
workletValueSetter
[native code]
terminating with uncaught exception of type facebook::jsi::JSError: undefined is not an object (evaluating 'delete animationCopy.callback')
decorateAnimation
[native code]
_f
[native code]
workletValueSetter
[native code]
CoreSimulator 732.18.0.2 - Device: iPhone 11 (FA4BBEC8-B78A-47A2-A8D6-C4273AB9E150) - Runtime: iOS 14.2 (18B79) - DeviceType: iPhone 11
```

gz#9961

**Comment by nhannah** [comment]
@ericlifs please use Reanimated 2.0.0-rc.0, it is noted in the docs that currently that is the version we support. Currently that is the latest version that is supported by expo, and as we support expo we are currently waiting for them to update before we can: https://docs.expo.io/versions/latest/sdk/reanimated/ .

**Comment by ericlifs** [comment]
@nhannah Hey, thanks for answering! I'm not using expo, I created all my apps by using react-native-cli. The thing is that Reanimated 2.0.0-rc.0  doesn't support yarn workspaces (as you can see in the issues I posted on the issue description), and that's blocking me from using stream-chat-react-native v3.0.0 on my project that uses yarn workspaces and I had to rely on the v2.2.2.

Isn't there a way of using stream-chat-react-native with an older version of Reanimated?

**Comment by nhannah** [comment]
@ericlifs What I was trying to point out is that we can't upgrade to Reanimated rc.3 yet because we have expo support, so we are on rc.0 in the package until expo upgrades, or we drop expo support for a few version. Ranimated 1 -> 2 is a significant change so there is no backward compatibility as we have moved to the new worklet / sharedValue implementation and do not use the v1 declarative style. I see there are people working on the mono repo issue for Ranimated so I think that is where the solution will lie. Reanimated has backing of the community at large and Facebook / Shopify so I expect they will come to a solution.

**Comment by vishalnarkhede** [comment]
@ericlifs Could you please share with us package.json for your monorepo workspaces and root? Curious if you tried using `nohoist` on dependencies coming from stream-chat-react-native package.

**Comment by vishalnarkhede** [comment]
@ericlifs ping ^^

**Comment by ericlifs** [comment]
Hey @vishalnarkhede! Sorry I was OOO these days, I will take a look again now and let you know. I have not tried by using the `nohoist` yet.

**Comment by ericlifs** [comment]
@vishalnarkhede Finally I was able to install reanimated 2 within my monorepo setup, I had to modify the metro config and it started working. I don't know if you had the time to take a look at the designs I have sent you, since some of the things are not on the docs.

Thanks,

Eric.

**Comment by vishalnarkhede** [comment]
@ericlifs thats great to hear that its working. Would you mind sharing the changes that you made to make it work, it may help other developers as well :)

Regarding the design changes, we will put up our slack clone early next week, where you will find tons of customisation, quite similar to what you are trying achieve.

**Comment by ericlifs** [comment]
@vishalnarkhede I applied these same changes https://github.com/brunolemos/react-native-web-monorepo/commit/af2a79a60a8ff3136caaa75ea8a6b84e96567687 to my react-native app. 

Right now I'm facing a lot of other issues, for example, I'm trying to get rid of the out of the box MessageSimple component and start writting one from my own from scratch and I'm getting this error:

![image](https://user-images.githubusercontent.com/7376892/112507303-472cc480-8d6d-11eb-805b-9fc739902b17.png)

I don't want to use all of the `ComponentClass<MessageUIComponentProps...` boilerplate that it has, I only want to use my custom types and define the props that I need, and return any element that I want.

Thanks.

**Comment by vishalnarkhede** [comment]
@ericlifs could you show us what type are you assigning your custom message component?

**Comment by ericlifs** [comment]
@vishalnarkhede this is my entire MessageSimple component: 

![image](https://user-images.githubusercontent.com/7376892/112512117-d76d0880-8d71-11eb-90db-18c0433eb8f5.png)

**Comment by ericlifs** [comment]
@vishalnarkhede I have started using the provided `MessageUIComponentProps` instead of my `MessageSimpleProps` and I was able to remove the types error. Now I'm seeing another issue where even after editing my component, those changes are not reflected on the web.

**Comment by vishalnarkhede** [comment]
Perfect!! Closing this issue for now :)
